src/libs/system-logging.ts

```diff
@@ -38,38 +42,101 @@ const saveLogEventInDB = (event: LogEvent): void => {
+      try {
+        msg = m.toString()
+      } catch {
+        msg = ''
+      }
+      if (msg !== '') {
+        saveLogEventInDB({ epoch: new Date().getTime(), level: 'error', msg: msg })
+      }
+    })
+  }
 
-const oldConsoleWarn = window.console.warn
-window.console.warn = (o) => {
-  oldConsoleWarn(o)
-  saveLogEventInDB({ epoch: new Date().getTime(), level: 'warn', msg: o })
-}
+  const oldConsoleWarn = window.console.warn
+  window.console.warn = (...o: any[]) => {
+    oldConsoleWarn(...o)
+    o.forEach((m) => {
+      let msg = m
+      try {
+        msg = m.toString()
+      } catch {
+        msg = ''
+      }
+      if (msg !== '') {
+        saveLogEventInDB({ epoch: new Date().getTime(), level: 'warn', msg: msg })
+      }
+    })
+  }
 
-const oldConsoleInfo = window.console.info
-window.console.info = (o) => {
-  oldConsoleInfo(o)
-  saveLogEventInDB({ epoch: new Date().getTime(), level: 'info', msg: o })
-}
+  const oldConsoleInfo = window.console.info
+  window.console.info = (...o: any[]) => {
+    oldConsoleInfo(...o)
+    o.forEach((m) => {
+      let msg = m
+      try {
+        msg = m.toString()
+      } catch {
+        msg = ''
+      }
+      if (msg !== '') {
+        saveLogEventInDB({ epoch: new Date().getTime(), level: 'info', msg: msg })
+      }
+    })
+  }
 
-const oldConsoleDebug = window.console.debug
-window.console.debug = (o) => {
-  oldConsoleDebug(o)
-  saveLogEventInDB({ epoch: new Date().getTime(), level: 'debug', msg: o })
-}
+  const oldConsoleDebug = window.console.debug
+  window.console.debug = (...o: any[]) => {
+    oldConsoleDebug(...o)
+    o.forEach((m) => {
+      let msg = m
+      try {
+        msg = m.toString()
+      } catch {
+        msg = ''
+      }
+      if (msg !== '') {
+        saveLogEventInDB({ epoch: new Date().getTime(), level: 'debug', msg: msg })
+      }
+    })
+  }
 
-const oldConsoleTrace = window.console.trace
-window.console.trace = (o) => {
-  oldConsoleTrace(o)
-  saveLogEventInDB({ epoch: new Date().getTime(), level: 'trace', msg: o })
-}
+  const oldConsoleTrace = window.console.trace
+  window.console.trace = (...o: any[]) => {
+    oldConsoleTrace(...o)
+    o.forEach((m) => {
+      let msg = m
+      try {
+        msg = m.toString()
+      } catch {
+        msg = ''
+      }
+      if (msg !== '') {
+        saveLogEventInDB({ epoch: new Date().getTime(), level: 'trace', msg: msg })
+      }
+    })
+  }
 
-const oldConsoleLog = window.console.log
-window.console.log = (o) => {
-  oldConsoleLog(o)
-  saveLogEventInDB({ epoch: new Date().getTime(), level: 'Log', msg: o })
+  const oldConsoleLog = window.console.log
+  window.console.log = (...o: any[]) => {
+    oldConsoleLog(...o)
+    o.forEach((m) => {
+      let msg = m
+      try {
+        msg = m.toString()
+      } catch {
+        msg = ''
+      }
+      if (msg !== '') {
+        saveLogEventInDB({ epoch: new Date().getTime(), level: 'Log', msg: msg })
+      }
+    })
```

Comment:
can we add a function to remove all this duplicated code and abstract this msg handle ?

Reply:
Sure!